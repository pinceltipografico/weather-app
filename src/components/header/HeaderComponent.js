import React, { Component } from 'react';
import logo from '../../assets/logo.svg';
import styles from './HeaderComponent.scss';
import PropTypes from 'prop-types';
import { Selects } from '../select/Selects';
import { citiesList } from '../../constants/citieslist';
import { WeatherConsumer } from '../../contexts/weatherContext';


class HeaderComponent extends Component {

  static propTypes = {
    title: PropTypes.string.isRequired
  }

  render () {
    return (
    <header className={styles.header}>
      <img src={logo} alt="logo" className={styles.logo} />  
      <h2 className={styles.title}>{this.props.title}</h2>
      
      <WeatherConsumer>
        { ({ fetchWeathers }) => (
          <Selects onChange={(value) => fetchWeathers(value)}
            label="Select a city"
            list={citiesList}/>
        )}
      </WeatherConsumer>
    </header>
    )
  }
}

export {
  HeaderComponent
}