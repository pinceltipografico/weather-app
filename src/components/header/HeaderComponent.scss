@import '~@/scss/mixins';

$white: #fff;

.header {
  @include flex-container;

  background-color: color(primary);
  flex: 1 1 auto;
  height: rem(100px);
  padding: 0 rem(20px);
  position: fixed;
  z-index: 2;
}

.logo {
  max-height: rem(80px);
}

.title {
  border-left: 1px dashed $white;
  border-right: 1px dashed $white;
  color: $white;
  margin: rem(10px) 0 0 rem(20px);
  padding: 0 rem(20px);
  
  @include breakpoint(phone-wide, max) {
    display: none;
  }
}
