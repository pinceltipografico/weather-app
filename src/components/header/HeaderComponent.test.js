import React from 'react'
import {shallow, mount} from 'enzyme'
import { HeaderComponent } from './HeaderComponent';
describe('HeaderComponent', () => {

  it('should render without problems', () => {
    const compoent = shallow(<HeaderComponent title="Weather app"/>)
    expect(compoent).toMatchSnapshot()
  })

  it('should have a logo image if proper src', () => {
    const compoent = mount(<HeaderComponent title="weather app" />)
    const img = compoent.find('img').first()
    expect(img.length).toBe(1)
    expect(img.getElement().props.src).toEqual('logo.svg')
  })

  it('should have a title tag', () => {
    const compoent = mount(<HeaderComponent title="weather app" />)
    const h2 = compoent.find('h2')
    expect(h2.length).toBe(1)
  })

})