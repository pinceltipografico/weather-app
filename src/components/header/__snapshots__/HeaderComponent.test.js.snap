// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should render without problems 1`] = `
<header>
  <img
    alt="logo"
    src="logo.svg"
  />
  <h2>
    Weather app
  </h2>
  <ContextConsumer>
    <Component />
  </ContextConsumer>
</header>
`;
